### PR TITLE
Skip unsupported and incompatible tests

### PR DESCRIPTION
**Pull request by moyodiallo**
The `skip` (builtin action) of `ocamltest` work well for skipping the unsupported and incompatible tests.
This PR can be the first step to replace the disable option added in the [Makefile (line 181-188)](https://github.com/ocaml-multicore/ocaml-multicore/blob/c692b7c0db8a48242876a6f32980b4403f304137/testsuite/Makefile#L181)

### CODE REVIEW

File: testsuite/disabled [hunk 1]
```diff
@@ -7,32 +7,38 @@
...
-tests/callback/'signals_alloc.ml' with 1.2 (native)
+#tests/callback/'signals_alloc.ml' with 1.1 (bytecode)
+#tests/callback/'signals_alloc.ml' with 1.2 (native)
 
+<<<<<<< HEAD
+# TODO: do we want to support the runtime naked pointer checker? (#9956)
+#runtime-naked-pointers
+
+=======
+>>>>>>> 5.00
```
Comment (kayceesrk) [inline testsuite/disabled]: Looks like this is left over from git merge.

**Comment by kayceesrk** [comment]
With these skipped tests, we can replace our modified testsuite/Makefile with the one from ocaml/ocaml trunk. Is that right? Other than that, what else is left to do in this PR?

**Comment by kayceesrk** [comment]
It is also useful to note that `make parallel` works with this PR.

**Comment by moyodiallo** [comment]
> It is also useful to note that `make parallel` works with this PR.

True it's works fine, this is the summary of `make parallel`:
```
Summary:
  3038 tests passed
   69 tests skipped
    0 tests failed
    0 tests disabled
  177 tests not started (parent test skipped or failed)
    0 unexpected errors
  3284 tests considered
```

**Comment by moyodiallo** [comment]
> With these skipped tests, we can replace our modified testsuite/Makefile with the one from ocaml/ocaml trunk. Is that right? Other than that, what else is left to do in this PR?

I replace the the testsuite/Makefile with ocaml/ocaml trunk one, and I test those commands: `make all-%`, `make one DIR=test/%`, `make parallel` and `make all`. All those commands work well. 

I think, it's ready to merge

**Comment by kayceesrk** [comment]
The Github workflows need to be updated. Currently, they're all doing `make all-enabled`: https://github.com/ocaml-multicore/ocaml-multicore/blob/35dfa8b4840c344bda8d219f05e39fec14e27814/tools/ci/actions/runner.sh#L68

**Comment by kayceesrk** [comment]
This is very nice @moyodiallo. I've approved these changes. I'll let @Engil have a quick look and merge.

**Review by kayceesrk — APPROVED** [review]

**Comment by abbysmal** [comment]
LGTM, great job, merging!

**Pull request merged**